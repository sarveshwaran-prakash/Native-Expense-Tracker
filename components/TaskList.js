import React, { useEffect, useState } from "react";
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  Dimensions,
  ScrollView,
} from "react-native";

const { width } = Dimensions.get("window");
const TaskList = ({ tasks, handleTaskOptionPress }) => {
  useEffect(() => {
    // Log tasks whenever tasks change for debugging
    console.log("Updated tasks:", tasks);
  }, [tasks]);
  return (
    <ScrollView contentContainerStyle={styles.container}>
      {tasks.map((task) => (
        <View key={task.id} style={styles.task}>
          <View style={styles.taskContent}>
            <Text numberOfLines={1} style={styles.taskTitle}>
              {task.title}
            </Text>
            {task.description && (
              <Text numberOfLines={1} style={styles.taskDescription}>
                {task.description}
              </Text>
            )}
          </View>
          <TouchableOpacity onPress={() => handleTaskOptionPress(task)}>
            <Text style={styles.optionText}>Options</Text>
          </TouchableOpacity>
        </View>
      ))}
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    width: "100%",
    paddingBottom: 20,
  },
  task: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    marginBottom: 10,
    padding: 10,
    borderWidth: 1,
    borderColor: "#ccc",
    borderRadius: 5,
    width: width * 0.9,
    alignSelf: "center",
  },
  taskContent: {
    flex: 1,
    marginRight: 10,
  },
  taskTitle: {
    fontWeight: "bold",
    marginBottom: 5,
  },
  taskDescription: {
    fontSize: 14,
    color: "gray",
  },
  optionText: {
    marginRight: 10,
  },
});

export default TaskList;
