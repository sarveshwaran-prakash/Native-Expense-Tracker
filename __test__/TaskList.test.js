import React from "react";
import renderer from "react-test-renderer";
import TaskList from "../components/TaskList";

describe("TaskList component", () => {
  const mockTasks = [
    { id: 1, title: "Task 1", description: "Description 1" },
    { id: 2, title: "Task 2", description: "Description 2" },
  ];

  it("renders tasks correctly", () => {
    const tree = renderer
      .create(<TaskList tasks={mockTasks} handleTaskOptionPress={() => {}} />)
      .toJSON();

    mockTasks.forEach((task) => {
      expect(tree).toMatchSnapshot();
    });
  });
});
