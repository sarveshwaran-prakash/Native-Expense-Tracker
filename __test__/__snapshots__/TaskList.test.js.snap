// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList component renders tasks correctly 1`] = `
<View
  contentContainerStyle={
    {
      "flexGrow": 1,
      "paddingBottom": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "padding": 10,
        "width": 675,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "marginRight": 10,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          {
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }
      >
        Task 1
      </Text>
      <Text
        numberOfLines={1}
        style={
          {
            "color": "gray",
            "fontSize": 14,
          }
        }
      >
        Description 1
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "marginRight": 10,
          }
        }
      >
        Options
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "padding": 10,
        "width": 675,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "marginRight": 10,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          {
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }
      >
        Task 2
      </Text>
      <Text
        numberOfLines={1}
        style={
          {
            "color": "gray",
            "fontSize": 14,
          }
        }
      >
        Description 2
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "marginRight": 10,
          }
        }
      >
        Options
      </Text>
    </View>
  </View>
</View>
`;

exports[`TaskList component renders tasks correctly 2`] = `
<View
  contentContainerStyle={
    {
      "flexGrow": 1,
      "paddingBottom": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "padding": 10,
        "width": 675,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "marginRight": 10,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          {
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }
      >
        Task 1
      </Text>
      <Text
        numberOfLines={1}
        style={
          {
            "color": "gray",
            "fontSize": 14,
          }
        }
      >
        Description 1
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "marginRight": 10,
          }
        }
      >
        Options
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "padding": 10,
        "width": 675,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "marginRight": 10,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          {
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }
      >
        Task 2
      </Text>
      <Text
        numberOfLines={1}
        style={
          {
            "color": "gray",
            "fontSize": 14,
          }
        }
      >
        Description 2
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "marginRight": 10,
          }
        }
      >
        Options
      </Text>
    </View>
  </View>
</View>
`;
